### PR TITLE
Full Site Editing:  Disable Move to Trash for templates and template types.

### LINKED ISSUE #34424: FSE: Trashed templates are applied to pages
It seems that we are not stripping out the trashed templates when getting the global template to be applied to pages without a specific template.

#### Steps to reproduce
1. Go to Templates (`/wp-admin/edit.php?post_type=wp_template`)
2. Select and move all of them to trash.
3. Go to Pages > Add new (`/wp-admin/post-new.php?post_type=page`)
4. Select any starter template.

#### What I expected

To see the editor without any template applied:

image

#### What happened instead

The trashed default template has been applied:

image
-->

### PR DESCRIPTION
**Pull request by rodrigoi**
#### Changes proposed in this Pull Request

* Removes the _Move to Trash_ button from the block editor when editing a template.
* Removes the _Trash_ quick action and the _Move to Trash_ bulk action on WP-Admin's edit list.
* Removes the _Delete_ quick and bulk actions on WP-Admin's taxonomy edit page.

| Before | After |
| - | - |
| before | after |
| before | after |
| before | after |

##### Block Editor

* Edit a Page
* Click the _Edit_ button on a template part.
* If closed, open the _Settings_ sidebar panel.
* Verify that the _Move to Trash_ button is hidden.

##### Templates List

* Navigate to `/wp-admin/edit.php?post_type=wp_template`
* Verify that the _Move to Trash_ option is missing from the _Bulk Actions_ dropdown.
* Verify that the _Trash_ link is missing from the row actions on the templates list.

##### Template Types

* Navigate to `/wp-admin/edit-tags.php?taxonomy=wp_template_type`
* Verify that the _Delete_ option is missing from the _Bulk Actions_ dropdown.
* Verify that the _Delete_ link is missing from the row actions on the template types list.

Fixes #34424

### CODE REVIEW

**Comment by Copons** [comment]
I was wondering if there is a simpler way to achieve this, and stumbled upon [this answer on Stack Overflow](https://wordpress.stackexchange.com/a/240180) that solves the same problem by removing the capability of deleting a CPT from various user roles.

I haven't tried it, but it sounds like a clever trick, and I guess it could also be extended to leave the capability untouched for super admins, which would be useful for testing and debugging.

File: apps/full-site-editing/full-site-editing-plugin/full-site-editing/editor/disable-trash-template/index.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/* global fullSiteEditing */
+
+/**
+ * External dependencies
+ */
+import domReady from '@wordpress/dom-ready';
+import { select, subscribe } from '@wordpress/data';
+
+domReady( () => {
```
Comment (mattwiebe) [inline apps/full-site-editing/full-site-editing-plugin/full-site-editing/editor/disable-trash-template/index.js]: We can hide this much more easily with just CSS:

```css
.post-type-wp_template .editor-post-trash {
  display: none;
}
```

Comment (gwwar) [reply apps/full-site-editing/full-site-editing-plugin/full-site-editing/editor/disable-trash-template/index.js]: Note that we can more cleanly override actions too, see https://github.com/Automattic/wp-calypso/pull/34855/files

Maybe a combo of display none, and adding a no-op for trashPost for this CPT?

**Comment by gwwar** [comment]
Thanks @rodrigoi I'm going to give tentative approval ✅, but would request changes to the block editor implementation as noted in https://github.com/Automattic/wp-calypso/pull/35278#discussion_r312660678 in either this PR or an immediate follow up

Screen Shot 2019-08-09 at 3 11 56 PM
Screen Shot 2019-08-09 at 3 12 21 PM
Screen Shot 2019-08-09 at 3 14 43 PM

File: apps/full-site-editing/full-site-editing-plugin/full-site-editing/editor/disable-trash-template/index.js (hunk 1 above)
Comment (rodrigoi) [reply apps/full-site-editing/full-site-editing-plugin/full-site-editing/editor/disable-trash-template/index.js]: Updated on a5cea6b77166fc59577ef8da014a4208f8b59cad.

**Pull request merged**